### PR TITLE
Simplifies module loading, uses commonjs everywhere

### PR DESCRIPTION
**Pull request by joemfb**
This PR makes argon2-wasm work in the browser and node by 

- always base64-encoding the WASM binary
- using MODULARIZE with the emscripten build
- using commonjs conventions everywhere
- generally simplified the initialization process

This means the package doesn't work directly in the browser, but must first be bundled/browserify'd in some manner. I think this is the simplest way to maintain javascript libraries, and I'm just looking to make this work reliably.

I've therefore removed the obsolete samples in `docs`. They could be restored if someone wants to add a build step to this repository.

The base64-encoding is kind-of gross, but simple and reliable. It should perhaps be replaced with either an asset-bundler in a build step, or some more robust fallback mechanisms. I can't find any good conventions for isomorphic libraries that include WASM.

closes #1

/cc @gavinpatkinson @c-johnson @fang-

### CODE REVIEW

**Comment by joemfb** [comment]
closes urbit/keygen-js#4

**Comment by c-johnson** [comment]
This is gorgeous. 

+1, LGTM, etc.

File: index.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+'use strict'
+
+if (!WebAssembly) {
+    throw new Error('WebAssembly not supported here.');
+}
+
+var loadModule = require('./generated/argon2.js');
```
Comment (c-johnson) [inline index.js]: Only rec I have is naming this something like "emscriptenWasmRunner", or something slightly more descriptive.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+// const GB = 1024 * MB;
+// const WASM_PAGE_SIZE = 64 * 1024;
+
+// const totalMemory = (2*GB - 64*KB) / 1024 / WASM_PAGE_SIZE;
+// const mem = args.mem || +(1024);
+// const initialMemory = Math.min(Math.max(Math.ceil(mem * 1024 / WASM_PAGE_SIZE), 256) + 256, totalMemory);
+// const wasmMemory = new WebAssembly.Memory({
+//     initial: initialMemory,
+//     maximum: totalMemory
+// });
```
Comment (joemfb) [inline index.js]: @Fang-, were these settings supposed to be passed to the emscripten module runner? Or do the memory arguments to the emscripten build suffice?

Comment (Fang-) [reply index.js]: /shrugs  
The original [passes this in as part of the `Module`](https://github.com/antelle/argon2-browser/blob/master/docs/js/calc.js#L96), but if this works without that here then maybe this is deprecated and/or taken care of for us by emscripten.

**Review by Fang- — APPROVED** [review]
Nothing about this jumps out at me, so looks good. Praise be!

File: index.js (hunk 2 above)
Comment (joemfb) [reply index.js]: Ok, thanks. I'll drop it.

**Pull request merged**